internal/app/carrier/repository.go

```diff
@@ -35,3 +41,42 @@ func (r *repository) InsertCarrierRequest(ctx context.Context, request model.Car
 	}
 	return nil
 }
+
+func (r *repository) UpdateCarrierRequest(ctx context.Context, parcel model.CarrierRequest, acceptStatus int, rejectStatus int, parcelStatus int, sourceTime time.Time) error {
+	//starting db transaction
+	tx, err := r.db.BeginTx(ctx, nil)
+	if err != nil {
+		log.Error().Err(err).Msg("[UpdateCarrierStatus] has failed to begin transactions.")
+		return fmt.Errorf("%v :%w", err, model.ErrTransaction)
+	}
+	//accept status update for carrier request table
+	if _, err = tx.ExecContext(ctx, updateAcceptQuery, acceptStatus, parcel.ParcelID, parcel.CarrierID); err != nil {
+		tx.Rollback()
+		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == errUniqueViolation {
+			log.Error().Err(err).Msg("[UpdateCarrierRequest] has failed to update status of carrier_rquest table to accept.")
+			return fmt.Errorf("%v :%w", err, model.ErrInvalid)
+		}
+		return err
+	}
+	if _, err := tx.ExecContext(ctx, updateRejectQuery, rejectStatus, parcel.ParcelID, parcel.CarrierID); err != nil {
+		tx.Rollback()
+		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == errUniqueViolation {
+			log.Error().Err(err).Msg("[UpdateCarrierRequest] has failed to update reject status for carrier_request table.")
+			return fmt.Errorf("%v :%w", err, model.ErrInvalid)
+		}
+		return err
+	}
+	if _, err := tx.ExecContext(ctx, updateParcelStatus, parcel.CarrierID, parcelStatus, sourceTime, parcel.ParcelID); err != nil {
+		tx.Rollback()
+		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == errUniqueViolation {
```

Comment:
why do we need unique key violation at here? can't imagine any scenario where unique key can be violated